src/connection.js

```diff
@@ -585,8 +594,12 @@ class Connection extends EventEmitter {
 
   requestTimeout() {
     this.requestTimer = undefined;
-    this.messageIo.sendMessage(TYPE.ATTENTION);
-    return this.transitionTo(this.STATE.SENT_ATTENTION);
+    this.cancel();
```

Comment:
Great, thanks for the help! I'm thinking its possible to still have the single cancel timer but to stick some bool flags to determine what kind of cancellation it is.